### PR TITLE
Convert DREAM to ask-and-tell

### PR DESCRIPTION
**Pull request by MichaelClerx**
See #219

### CODE REVIEW

**Comment by MichaelClerx** [comment]
This seems to work very well on the logistic problem at least. Happy to just merge it once the test pass @ben18785 @sanmitraghosh ?

**Comment by MichaelClerx** [comment]
This looks like it works in the example notebook, so I'd like to merge now and stick it in functional testing!

File: pints/_mcmc/_dream.py [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+
+        #
+        # Default settings
+        #
+
+        # Normal proposal std.
+        self._b = 0.01
+
+        # b* distribution for e ~ U(-b*, b*)
+        self._b_star = 0.01
```
Comment (martinjrobins) [inline pints/_mcmc/_dream.py]: do these numbers need to be hyper-parameters?

File: pints/_mcmc/_dream.py [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+
+        # Probability of longer gamma versus regular
+        self._p_g = 0.2
```
Comment (martinjrobins) [inline pints/_mcmc/_dream.py]: hyper-param?

File: pints/_mcmc/_dream.py [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+
+        # Determines maximum delta to choose in sums
+        self._delta_max = 3
```
Comment (martinjrobins) [inline pints/_mcmc/_dream.py]: hyper-param?

File: pints/_mcmc/_dream.py [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+
+        # Initial phase
+        self._initial_phase = True
+
+        # Variable or constant cross-over mode
+        self._constant_crossover = False
```
Comment (martinjrobins) [inline pints/_mcmc/_dream.py]: discrete hyper-param?

File: pints/_mcmc/_dream.py [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+
+        # Constant CR probability
+        self._CR = 0.5
```
Comment (martinjrobins) [inline pints/_mcmc/_dream.py]: another hyper-param?

File: pints/_mcmc/_dream.py [hunk 6]
```diff
@@ -0,0 +1,358 @@
...
+        # 1 / Variable crossover probability
+        self._nCR = 3
```
Comment (martinjrobins) [inline pints/_mcmc/_dream.py]: another hyper-param?

File: pints/_mcmc/_dream.py [hunk 7]
```diff
@@ -0,0 +1,358 @@
...
+        """
+        self._constant_crossover = True if enabled else False
+
+    def set_hyper_parameters(self, x):
+        """
+        The hyper-parameter vector is ``[b]``.
+
+        See :meth:`TunableMethod.set_hyper_parameters()`.
+        """
+        self.set_b(x[0])
```
Comment (martinjrobins) [inline pints/_mcmc/_dream.py]: b is the only hyper-parameter listed...

**Comment by MichaelClerx** [comment]
Good point! `b` is the only parameter @ben18785 stuck a get/set in for. I don't know the method well enough to say anything about the others!

**Comment by martinjrobins** [comment]
its ok to merge for now, I'll put in an issue for the rest

**Comment by MichaelClerx** [comment]
Cheers!

**Pull request merged**